Objects/genericaliasobject.c

```diff
@@ -156,13 +156,21 @@ ga_repr(PyObject *self)
     return NULL;
 }
 
-// isinstance(obj, TypeVar) without importing typing.py.
-// Returns -1 for errors.
-static int
-is_typevar(PyObject *obj)
+/* Checks if a variable number of names are from typing.py.
+*  If any one of the names are found, return 1, else 0.
+**/
+static inline int
+is_typing_names(PyObject *obj, int num, ...)
 {
+    va_list names;
+    va_start(names, num);
+
     PyTypeObject *type = Py_TYPE(obj);
-    if (strcmp(type->tp_name, "TypeVar") != 0) {
+    int cmp = 1;
+    for (int i = 0; i < num && cmp != 0; ++i) {
+        cmp &= strcmp(type->tp_name, va_arg(names, const char *));
```

Comment:
That would make checking for paramspec fail because the equivalent python code is (please pardon the minimal example):
```python
def is_typevar_like(obj):
    names = ('TypeVar', 'ParamSpec')
    for name in names:
        if x!= name:
            return False

# False
is_typevar_like('ParamSpec')
```
I used `&=` since 0 short-circuits (unsure if that's the correct term) the entire & operation to 0 as long as there's a single 0 (single match) in there. ``(-1) & 1`` will always give ``1``. That's the closest equivalent to python's `any` I could think of.

Reply:
Oh, I didn't see that your 'for' condition skips if `cmp` is zero. Sorry!

But that's kind of easy to overlook. How about this way:
```
    for (int i = 0; i < num; ++i) {
        if (strcmp(type->tp_name, va_arg(names, const char *)) == 0) {
            cmp = 0;
            break;
        }
    }
```
Then I would rename `cmp` to `eq` so the reader understands it represents "equality" or perhaps to `hit` meaning "we found the name in the list of names" ("hit" is somewhat traditional for a search loop, which this is).

Finally I would change the function name to `is_typing_name()` (i.e., singular). This again helps the reader get the purpose of the function.